src/main/java/hudson/plugins/audit_trail/LogFileAuditLogger.java

```diff
@@ -30,33 +48,143 @@ public class LogFileAuditLogger extends AuditLogger {
+    @Deprecated
     public LogFileAuditLogger(String log, int limit, int count, String logSeparator) {
+        this(log, limit, count, logSeparator, false);
+    }
+
+    @DataBoundConstructor
+    public LogFileAuditLogger(String log, int limit, int count, String logSeparator, boolean rotateDaily) {
         this.log = Util.replaceMacro(log, EnvVars.masterEnvVars);
-        this.limit = limit;
+        this.basePattern = Paths.get(log);
+        this.limit = rotateDaily ? 0 : limit;
         this.count = count;
         this.logSeparator = Optional.ofNullable(logSeparator).orElse(DEFAULT_LOG_SEPARATOR);
+        this.rotateDaily = rotateDaily;
+
+        if (rotateDaily) {
+            initializeDailyRotation();
+        } else {
+            configure();
+        }
+    }
+
+    /**
+     * Initializes initInstant to the instant obtained from the latest daily
+     * log file saved on disk (if present), or if not present, to the current instant.
+     */
+    private void initializeDailyRotation() {
+        Path directoryPath = basePattern.getParent();
+        boolean directoryExists = false;
+        if (directoryPath != null) {
+            directoryExists = directoryPath.toFile().exists();
+        }
+
+        if (directoryExists) {
+            // audit-log.log-2022-10-19-15-50.0
+            Collection<File> files = FileUtils.listFiles(new File(directoryPath.toString()), new RegexFileFilter(".*" + FilenameUtils.getName(basePattern.toString()) + DAILY_ROTATING_FILE_REGEX_PATTERN), DirectoryFileFilter.DIRECTORY);
+            if (files.size() > 0) {
+                List<File> orderedList = files.stream().sorted(LASTMODIFIED_REVERSE).collect(Collectors.toList());
+                File lastFile = orderedList.get(0);
+                Matcher matcher = Pattern.compile("[0-9]{4}-[0-9]{2}-[0-9]{2}").matcher(lastFile.getName());
+                if (matcher.find()) {
+                    // Initialize initInstant with the date saved on the audit file name
+                    // See example https://stackoverflow.com/questions/14385834/java-regex-group-0
+                    initInstant = Instant.parse(matcher.group(0) + "T00:00:00.00Z").atZone(ZoneId.systemDefault());
+                }
+            }
+        }
+        // Initialize initInstant to the current instant
+        if (initInstant == null) {
+            initInstant = ZonedDateTime.now().truncatedTo(ChronoUnit.DAYS);
+        }
         configure();
     }
 
+    private boolean shouldRotate() {
+        return ZonedDateTime.now().isAfter(initInstant.plus(Duration.ofDays(1)));
+    }
+
+    /**
+     * Rotates the daily rotation logger
+     */
+    private void rotate() {
+        if (handler != null) {
+            handler.close();
+        }
+        initInstant = ZonedDateTime.now().truncatedTo(ChronoUnit.DAYS);
+        configure();
+        // After rotation we remove old files if needed
+        removeOldFiles();
+    }
+
+    protected String computePattern() {
```

Comment:
nit: can be package protected:

```suggestion
    String computePattern() {
```